src/main/java/edu/ucla/library/services/metadata/MetadataSetter.java

```diff
@@ -0,0 +1,350 @@
+
+  /**
+   * File/directory doesn't exist.
+  */
+    public static final int FILE_DOESNT_EXIST = 101;
+
+  /**
+   * ffprobe executable doesn't exist.
+  */
+    public static final int PROBE_DOESNT_EXIST = 102;
+
+  /**
+   * Constant for width column name.
+  */
+    private static final String HEADER_WIDTH = "media.width";
+
+  /**
+   * Constant for height column name.
+  */
+    private static final String HEADER_HEIGHT = "media.height";
+
+  /**
+   * Constant for duration column name.
+  */
+    private static final String HEADER_DURATION = "media.duration";
+
+  /**
+   * Constant for format column name.
+  */
+    private static final String HEADER_FORMAT = "media.format";
+
+  /**
+   * Constant for column position of media.width.
+  */
+    private static final int WIDTH_OFFSET = 4;
+
+  /**
+   * Constant for column position of media.height.
+  */
+    private static final int HEIGHT_OFFSET = 3;
+
+  /**
+   * PConstant for column position of media.duration.
+  */
+    private static final int DURATION_OFFSET = 2;
+
+  /**
+   * Constant for column position of media.format.
+  */
+    private static final int FORMAT_OFFSET = 1;
+
+
+  /**
+   * Path to CSV file (or directory of CSV files) to be updated.
+  */
+    @Parameters(index = "0", description = "The CSV file/directory to process.")
+    private static String CSV_PATH;
+
+  /**
+   * Path to media files to be read.
+  */
+    @Parameters(index = "1", description = "The parent directory/mount point holding media files;"
+        + " this is prepended to file name in CSV.  If file name in CSV is /ephraim/audio/audio.wav, and"
+        + " full path to audio.wav is /mount/dlcs/masters/ephraim/audio/audio.wav, enter /mount/dlcs/masters/")
+    private static String MEDIA_PATH;
+
+  /**
+   * Path to ffmpeg probe utility.
+  */
+    @Parameters(index = "2", description = "The path to ffprobe executable.")
+    private static String FFMPEG_PATH;
+
+  /**
+   * Path where modified CSV file(s) will be written.
+  */
+    @Parameters(index = "3", description = "The directory where output file(s) are written.")
+    private static String OUTPUT_PATH;
+
+  /**
+   * Private constructor for MetaDataSetter class.
+  */
+    private MetadataSetter() {
+    }
+
+  /**
+   * Main method for command-line execution.
+   *
+   * @param args array of parameters
+  */
+    @SuppressWarnings("uncommentedmain")
+    public static void main(final String[] args) {
+        final int exitCode = new CommandLine(new MetadataSetter()).execute(args);
+        System.exit(exitCode);
+    }
+
+  /**
+   * Callable method executed by picoli framework.
+   *
+  */
+    @Override
+    public Integer call() {
+        if (!fileDirExists(CSV_PATH)) {
+            return FILE_DOESNT_EXIST;
+        }
+        if (!fileDirExists(MEDIA_PATH)) {
+            return FILE_DOESNT_EXIST;
+        }
+        if (!fileDirExists(OUTPUT_PATH)) {
+            return FILE_DOESNT_EXIST;
+        }
```

Comment:
It appears this conditional is still causing invocation to exit if the output dir doesn't exist.

Reply:
Silly me put the "create directories" code in a different method...